### PR TITLE
Add manylinux builder

### PR DESCRIPTION
**Pull request by irexyc**
add build scripts for docker and whl package.

### CODE REVIEW

File: builder/manywheel/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# lmdeploy/builder/manywheel
```
Comment (lvhan028) [inline builder/manywheel/README.md:1]: How about `# Build lmdeploy wheel`?

File: builder/manywheel/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+To build all docker images you can use the convenience script:
+
+```bash
+./build_all_docker.sh
+# Build with pushing
+WITH_PUSH=true ./build_all_docker.sh
+```
+
+To build a specific docker image use:
```
Comment (lvhan028) [inline builder/manywheel/README.md:13]: `To build a docker image with specific cuda version or manylinux-docker version, you may use: `

File: builder/manywheel/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# lmdeploy/builder/manywheel
+
+## Building docker images
+
+To build all docker images you can use the convenience script:
```
Comment (lvhan028) [inline builder/manywheel/README.md:5]: convenient

File: builder/manywheel/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+WITH_PUSH=true ./build_all_docker.sh
+```
+
+To build a specific docker image use:
+
+```bash
+MANY_LINUX_VERSION=2014 GPU_ARCH_VERSION=11.8 ./build_docker.sh
+```
+
+## Building lmdeploy
```
Comment (lvhan028) [inline builder/manywheel/README.md:19]: Build lmdeploy wheel

File: builder/manywheel/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# lmdeploy/builder/manywheel
+
+## Building docker images
```
Comment (lvhan028) [inline builder/manywheel/README.md:3]: Prepare docker image

File: builder/manywheel/entrypoint_build.sh [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+export USERID=${USERID}
+export GROUPID=${GROUPID}
+
+source /opt/conda/bin/activate
+conda activate $PYTHON_VERSION
+
+git clone https://github.com/InternLM/lmdeploy
+cd lmdeploy
+mkdir build && cd build
+cmake .. \
```
Comment (lvhan028) [inline builder/manywheel/entrypoint_build.sh:15]: may use `generate.sh` instead

File: builder/manywheel/build_docker.sh [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+DOCKER_TAG=cuda${GPU_ARCH_VERSION}
+
+DOCKER_IMAGE=openmmlab/lmdeploy-builder:${DOCKER_TAG}
+if [[ -n ${MANY_LINUX_VERSION} ]]; then
+    DOCKERFILE_SUFFIX=_${MANY_LINUX_VERSION}
+else
+    DOCKERFILE_SUFFIX=''
+fi
+
+(
```
Comment (lvhan028) [inline builder/manywheel/build_docker.sh:21]: may remove parentheses

File: builder/manywheel/scripts/install_cuda.sh [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+function install_118 {
+    echo "Installing CUDA 11.8 and cuDNN 8.7 and NCCL 2.15"
+    rm -rf /usr/local/cuda-11.8 /usr/local/cuda
+    # install CUDA 11.8.0 in the same container
+    wget -q https://developer.download.nvidia.com/compute/cuda/11.8.0/local_installers/cuda_11.8.0_520.61.05_linux.run
+    chmod +x cuda_11.8.0_520.61.05_linux.run
+    ./cuda_11.8.0_520.61.05_linux.run --toolkit --silent
+    rm -f cuda_11.8.0_520.61.05_linux.run
+    rm -f /usr/local/cuda && ln -s /usr/local/cuda-11.8 /usr/local/cuda
```
Comment (lvhan028) [inline builder/manywheel/scripts/install_cuda.sh:13]: `usr/local/cuda` has been removed in line 7

**Review by lvhan028 — APPROVED** [review]
LGTM

**Review by AllentDan — APPROVED** [review]
LGTM

**Pull request merged**